src/Entities/Player.cpp

```diff
@@ -89,6 +89,12 @@ cPlayer::cPlayer(cClientHandlePtr a_Client, const AString & a_PlayerName) :
 	m_UUID((a_Client != nullptr) ? a_Client->GetUUID() : ""),
 	m_CustomName("")
 {
+	if (a_PlayerName.empty() || a_PlayerName.length() > 16)
+	{
+		LOGERROR("Playername of %s is too long(>16) or empty", a_PlayerName.c_str());
+		VERIFY(false);
+	}
```

Comment:
You've got `ASSERT` and `VERIFY` mixed up. `ASSERT` only fires in debug mode and is compiled out in release mode, `VERIFY` is the same as `ASSERT` in debug mode and keep the code in its parameter in Release mode.
In this case it evaluates to `false;` which is a valid C++ expression saying "no operation".

Still, the proper thing to do is to assert if you expect this condition already handled elsewhere, or fix it silently if you expect to actually receive the bad data.